Remove log2 and pow declarations in neurocorrelates

alackles opened the pull request:
Fixes #264 by removing the line declaring log2 and pow as members of std namespace. `#include <math.h>` takes care of these declarations and the feature should work as before.

alackles commented:
Tested on Windows and Mac. Needs testing on Linux.

VincentRagusa commented:
Compiles successfully on Manjaro Linux with c++ (GCC) 8.2.1 20180831

cgnitash commented:
Actually, it's an incorrect header. <math.h> is a C library. <cmath> is the correct library, and that does have log2, etc, in the std namespace.

alackles commented:
Thanks for the fix @cgnitash. I've now replaced `#include <math.h>` with `#include <cmath>` and added back in the `using std::log; using std::pow;` line.

VincentRagusa commented:
Are the using statements necessary?

On Thu, Sep 13, 2018, 1:18 PM Acacia Ackles <notifications@github.com>
wrote:

> Thanks for the fix @cgnitash <https://github.com/cgnitash>. I've now
> replaced #include <math.h> with #include <cmath> and added back in the using
> std::log; using std::pow; line.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Hintzelab/MABE/pull/265#issuecomment-421084326>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AhoeifjvmKFPOjuWFhnDDXdUYIyudR28ks5uapNXgaJpZM4WnZHw>
> .
>

cgnitash commented:
@VincentRagusa not necessarily, but then you have to qualify the functions with std::

VincentRagusa commented:
I thought that was standard practice, to prepend std:: on everything and
avoid using 'using'.

On Thu, Sep 13, 2018, 1:47 PM cgnitash <notifications@github.com> wrote:

> @VincentRagusa <https://github.com/VincentRagusa> not necessarily, but
> then you have to qualify the functions with std::
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Hintzelab/MABE/pull/265#issuecomment-421093465>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AhoeiU8f0o1BG6CcWs_SShnt6ylY--Asks5uapo6gaJpZM4WnZHw>
> .
>

JorySchossau commented:
@VincentRagusa Yes, it is a decision we made and never wrote down (!) to always use `std::` but it is a gradual migration.
